lib/ansible/modules/cloud/spotinst/spotinst_aws_elastigroup.py

```diff
@@ -0,0 +1,1509 @@
+                          'target',
+                          'threshold']
+
+action_fields = ['action_type',
+                 'adjustment',
+                 'min_target_capacity',
+                 'max_target_capacity',
+                 'target',
+                 'minimum',
+                 'maximum']
+
+signal_fields = ['name',
+                 'timeout']
+
+multai_lb_fields = ['balancer_id',
+                    'project_id',
+                    'target_set_id',
+                    'az_awareness',
+                    'auto_weight']
+
+persistence_fields = ['should_persist_root_device',
+                      'should_persist_block_devices',
+                      'should_persist_private_ip']
+
+strategy_fields = ['risk',
+                   'utilize_reserved_instances',
+                   'fallback_to_od',
+                   'on_demand_count',
+                   'availability_vs_cost',
+                   'draining_timeout',
+                   'spin_up_time',
+                   'lifetime_period']
+
+ebs_fields = ['delete_on_termination',
+              'encrypted',
+              'iops',
+              'snapshot_id',
+              'volume_type',
+              'volume_size']
+
+bdm_fields = ['device_name',
+              'virtual_name',
+              'no_device']
+
+kubernetes_fields = ['api_server',
+                     'token']
+
+right_scale_fields = ['account_id',
+                      'refresh_token']
+
+rancher_fields = ['access_key',
+                  'secret_key',
+                  'master_host']
+
+chef_fields = ['chef_server',
+               'organization',
+               'user',
+               'pem_key',
+               'chef_version']
+
+az_fields = ['name',
+             'subnet_id',
+             'placement_group_name']
+
+opsworks_fields = ['layer_id']
+
+scaling_strategy_fields = ['terminate_at_end_of_billing_hour']
+
+mesosphere_fields = ['api_server']
+
+ecs_fields = ['cluster_name']
+
+multai_fields = ['multai_token']
+
+
+def handle_elastigroup(client, module):
+    has_changed = False
+    should_create = False
+    group_id = None
+    message = 'None'
+
+    name = module.params.get('name')
+    state = module.params.get('state')
+    uniqueness_by = module.params.get('uniqueness_by')
+    external_group_id = module.params.get('id')
+
+    if uniqueness_by == 'id':
+        if external_group_id is None:
+            should_create = True
+        else:
+            should_create = False
+            group_id = external_group_id
+    else:
+        groups = client.get_elastigroups()
+        group_found, group_id = find_group_with_same_name(groups, name)
+        if group_found is True:
+            should_create = False
+        else:
+            should_create = True
+
+    if should_create is True:
+        if state == 'present':
+            eg = expand_elastigroup(module, is_update=False)
+
+            processLog = open("/var/log/spotinst/ansible-module.log", 'a+')
```

Comment:
Instead of manually creating log files such as this, it's preferred to use the `module.debug(.........)` method so that the Ansible controller is in charge of deciding whether the logs should be displayed.

Reply:
👍